fa25-team-b/scripts/popularity_graph_generation.py

```
@@ -0,0 +1,669 @@
+        if use_log:
+            ax.set_yscale("log")
+            ax.yaxis.set_major_formatter(StrMethodFormatter("${x:,.0f}"))
+        else:
+            ax.yaxis.set_major_formatter(StrMethodFormatter("{x:,.1f}"))
+
+        ax.grid(True, axis="y", color="#E2E8F0", linewidth=0.8, alpha=0.7)
+        ax.spines["top"].set_visible(False)
+        ax.spines["right"].set_visible(False)
+
+        # Add count annotations - modify x-tick labels to include count
+        for i, cat in enumerate(categories):
```

Ruff (0.14.8)

Loop control variable `cat` not used within loop body

Rename unused `cat` to `_cat`

(B007)